ui: upgrade `antd` package to v5 in cluster-ui

koorosh opened the pull request:
This change upgrades `antd` package (in cluster-ui) up to 5.x version to match
the version in CC Console and make it easier to extract components.
It is preferable to review those changes per commit:
- first commit introduces changes in build files (package.json, webpack, etc)
- next commit contains changes produced by command:
 `pnpm --package=@ant-design/codemod-v5 dlx antd5-codemod src`
- final commit resolves braking changes after `antd` version upgrade.

Note, those changes doesn't fix broken styles of components and it affects 
how some components look in Db Console.

Resolves: https://cockroachlabs.atlassian.net/browse/CC-28016

- [x] mirror NPM dependencies with `dev ui mirror-deps`

koorosh committed:
ui: upgrade antd to 5.x version in cluster-ui

- upgrade and version to 5.6.1 in package.json. It is the same
version of antd that's used in CC Console.
- remove babel configs related to antd (deprecated)
- remove overridden antd global styles
- all of these changes are done according to migration
instruction https://ant.design/docs/react/migration-v5

Release note: None

koorosh committed:
ui: automatic fixes for antd migration 4.x -> 5.x in cluster-ui

Those changes produced by command:
`pnpm --package=@ant-design/codemod-v5 dlx antd5-codemod src`
which applies automatic fixes during Antd version upgrade to 5v.

Release note: None

koorosh committed:
ui: resolve conflicts in components that depend on Antd package in cluster-ui

This change resolves issues that couldn't be fixed automatically during
`antd` version upgrade and require changes to resolve breaking changes.
Following components were mostly affected:
- `<Icon/>`
- `<DatePicker/>`
- `<Select/>`
In addition, proper types were provided due to resolve compilation errors.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/123973)
<!-- Reviewable:end -->

koorosh committed:
ui: fix styles of DateRangeMenu component

- default button type with borders was overridden by global styles so
former were removed;
- time and date selectors size is set to 'large' to keep the same size of
component as before;

Release note: None

koorosh committed:
ui: fix Search component styling after antd version upgrade

This change adjusts styling of <Search /> component after
antd version upgrade that affected styles.
- cleaned up some style that override `antd` styles as they
don't make any sense;
- removed `<Form />` component as a wrapper that was responsible to
handle event when user presses Enter button. Now it is possible to
handle within `<Input />` component.

Release note: None

innabs commented:
UX concerns:
- SQL activity page, time range selector and search placeholder text
- Databases page, search input active state
- Metrics page time selector dropdown alignment in the input  + back/forward buttons

koorosh committed:
ui: add antd theme to cluster-ui package

This change adds a copy of antd theme from CC Console
to rely on the same styles. It will help to adjust
components to the same styles as in CC Console during
extraction components.

Release note: None

koorosh committed:
ui: re-work Modal and Diagnostics modal

ActivateStatementDiagnosticsModal component is based on pure
Antd components and has been dramatically affected after
upgrade of Antd version to 5.x.
The main issue that new components are based on flexbox styles
and it caused many changes in layout and how components
structured.
So this change introduced following changes:
- avoid using `<div>` elements for structuring layout, instead
use <Space> and <Grid> components.
- Remove unused styles and rely on antdTheme instead as much as
possible. There is some styles defined which cannot be defined with
theme.

Release note: None

koorosh committed:
ui: fix JobsProfileView component styles

This patch doesn't affect how `JobsProfileView` component
looks, instead it fixes some broken styles after `antd`
package has been upgraded. Also, `Space` components are
used for better layout structuring instead of `<Row>` and
`<Col>` components.

Release note: None

laurenbarker commented:
I checked this out and ran `mirror-deps` 👍

koorosh commented:
Closing this PR in favor of https://github.com/cockroachdb/cockroach/pull/124100 as changes in cluster-ui consequently break code in Db Console (incompatible interfaces of Antd components)
